components/nimbus/src/tests/test_behavior.rs

```diff
@@ -0,0 +1,657 @@
+        ));
+        assert!(matches!(
+            store
+                .events
+                .get(&"event-1".to_string())
+                .unwrap()
+                .intervals
+                .get(&Interval::Days)
+                .unwrap()
+                .data
+                .buckets
+                .len(),
+            3
+        ));
+        assert!(matches!(
+            store
+                .events
+                .get(&"event-1".to_string())
+                .unwrap()
+                .intervals
+                .get(&Interval::Days)
+                .unwrap()
+                .data
+                .buckets[0],
+            1
+        ));
+        assert!(matches!(
+            store
+                .events
+                .get(&"event-1".to_string())
+                .unwrap()
+                .intervals
+                .get(&Interval::Days)
+                .unwrap()
+                .data
+                .buckets[1],
+            0
+        ));
+        assert!(matches!(
+            store
+                .events
+                .get(&"event-1".to_string())
+                .unwrap()
+                .intervals
+                .get(&Interval::Days)
+                .unwrap()
+                .data
+                .buckets[2],
+            0
+        ));
+
+        assert!(matches!(
+            store
+                .events
+                .get(&"event-2".to_string())
+                .unwrap()
+                .intervals
+                .get(&Interval::Days)
+                .unwrap()
+                .data
+                .buckets
+                .len(),
+            1
+        ));
+        assert!(matches!(
+            store
+                .events
+                .get(&"event-2".to_string())
+                .unwrap()
+                .intervals
+                .get(&Interval::Days)
+                .unwrap()
+                .data
+                .buckets[0],
+            0
+        ));
+
+        Ok(())
+    }
+
+    #[test]
+    fn query_sum_should_function() -> Result<()> {
+        let counter1 = MultiIntervalCounter::new(vec![
+            SingleIntervalCounter::new(IntervalConfig::new(12, Interval::Months)),
+            SingleIntervalCounter::new(IntervalConfig::new(28, Interval::Days)),
+        ]);
+
+        let counter2 = MultiIntervalCounter::new(vec![
+            SingleIntervalCounter::new(IntervalConfig::new(12, Interval::Months)),
+            SingleIntervalCounter::new(IntervalConfig::new(28, Interval::Days)),
+        ]);
+
+        let mut store = EventStore::from(vec![
+            ("event-1".to_string(), counter1),
+            ("event-2".to_string(), counter2),
+        ]);
+
+        store.record_event("event-1".to_string(), Some(Utc::now() + Duration::days(2)))?;
+        store.record_event("event-1".to_string(), Some(Utc::now() + Duration::days(3)))?;
+        store.record_event("event-1".to_string(), Some(Utc::now() + Duration::days(3)))?;
+
+        assert!(matches!(
```

Comment:
You can also use `assert_eq!()` here and avoid the `matches!()`. This is absolutely fine if you prefer this, it's perfectly understandable as is.

Reply:
Ah I'll make those adjustments in my next PR -- I was reviewing something of James' earlier today and noticed it there as well!